src/ui/public/vislib/lib/axis.js

```diff
@@ -15,15 +15,17 @@ export default function AxisFactory(Private) {
   const AxisConfig = Private(AxisConfigProvider);
 
   class Axis extends ErrorHandler {
-    constructor(args) {
+    constructor(config, args) {
```

Comment:
Now that `args` is no longer all of the arguments, i think we should consider different names here. My initial approach would be to rename `config` to `visConfig` (it's an instance of `VisConfig` right?) and `args` to `userConfigVars` or something that describes its purpose a little better.